### PR TITLE
Make grantable configurable for initial privileges set

### PR DESCRIPTION
**Pull request by amiorin**
In Zalando we have a dedicated process to grant access to data. We don't want user to grant access data on their own. We also want to allow our user to use CTAS. The default behaviour of Hive is that if you create a table you can grant all privileges. This doesn't work for us because we have an internal process to grant access to data. In this patch I've added a new config entry to make grantable false.

I've also changed the SqlStandardAccessControl to stop requiring grantable for ``show create table`` if grantable is false in ``CREATE TABLE`` to be consistent with the new semantic.

A side note. We have noticed a problem with authorized views and list partitions. When users want to know if today partition was added they cannot use ``SELECT * FROM hive.web."page_views$partitions"`` because they have only access to the view. They are forced to run ``select distinct partition_column from view`` that is more expensive.

### CODE REVIEW

File: presto-hive/src/main/java/io/prestosql/plugin/hive/HiveConfig.java [hunk 1]
```diff
@@ -937,4 +938,17 @@ public HiveConfig setProjectionPushdownEnabled(boolean projectionPushdownEnabled
         this.projectionPushdownEnabled = projectionPushdownEnabled;
         return this;
     }
+
+    public boolean getGrantOption()
+    {
+        return grantOption;
+    }
+
+    @Config("hive.grant-option")
```
Comment (losipiuk) [inline presto-hive/src/main/java/io/prestosql/plugin/hive/HiveConfig.java:947]: IMO `hive.create-with-grant-option` would be more descriptive.

File: presto-hive/src/main/java/io/prestosql/plugin/hive/security/SqlStandardAccessControl.java [hunk 2]
```diff
@@ -154,8 +158,9 @@ public void checkCanShowSchemas(ConnectorSecurityContext context)
     @Override
     public void checkCanShowCreateTable(ConnectorSecurityContext context, SchemaTableName tableName)
     {
+        boolean grantOptionRequired = hiveConfig.getGrantOption();
         // This should really be OWNERSHIP, but Hive uses `SELECT with GRANT`
-        if (!checkTablePermission(context, tableName, SELECT, true)) {
+        if (!checkTablePermission(context, tableName, SELECT, grantOptionRequired)) {
```
Comment (losipiuk) [inline presto-hive/src/main/java/io/prestosql/plugin/hive/security/SqlStandardAccessControl.java:163]: This is unexpected. Why do we need grant option for `show crate table`. It seems we should just have `false` here.

File: presto-hive/src/main/java/io/prestosql/plugin/hive/HiveConfig.java (hunk 1 above)
Comment (kokosing) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/HiveConfig.java:947]: I think this be should defined only when `hive.security=sql` is turned on. That means that we might need `SqlStandardAccessControlConfig` that is only loaded in `SqlStandardSecurityModule`.

File: presto-hive/src/main/java/io/prestosql/plugin/hive/security/SqlStandardAccessControl.java (hunk 2 above)
Comment (kokosing) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/security/SqlStandardAccessControl.java:163]: I think this is how it is implemented in Hive. In general Presto requires ownership privilege for `SHOW CREATE...` statement.

Comment (amiorin) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/security/SqlStandardAccessControl.java:163]: Indeed

**Comment by amiorin** [comment]
After discovering some problems with this feature, we are now evaluating to write an "admission controller" (the name comes from K8s) inside our Presto proxy. It will parse the query and reject queries that are not compliant. We don't need this PR anymore. Thank you for the review.

**Pull request closed without merging**